src/app/api/routes/media.py

```diff
@@ -85,64 +86,101 @@ async def delete_media(media_id: int = Path(..., gt=0)):
 
 
 @router.post("/{media_id}/upload", response_model=MediaOut, status_code=200)
-async def upload_media(media_id: int = Path(..., gt=0),
-                       file: UploadFile = File(...),
-                       current_device: DeviceOut = Security(get_current_device, scopes=["device"])):
+async def upload_media(
+    background_tasks: BackgroundTasks,
+    media_id: int = Path(..., gt=0),
+    file: UploadFile = File(...),
+    current_device: DeviceOut = Security(get_current_device, scopes=["device"])
+):
     """
     Upload a media (image or video) linked to an existing media object in the DB
     """
-    entry = await check_for_media_existence(media_id, current_device.id)
+
+    # Check in DB
+    entry = await check_media_registration(media_id, current_device.id)
 
     # Concatenate the first 32 chars (to avoid system interactions issues) of SHA256 hash with file extension
-    file_name = f"{hash_content_file(file.file.read())[:32]}.{file.filename.rpartition('.')[-1]}"
+    file_hash = hash_content_file(file.file.read())
+    file_name = f"{file_hash[:32]}.{file.filename.rpartition('.')[-1]}"
     # Reset byte position of the file (cf. https://fastapi.tiangolo.com/tutorial/request-files/#uploadfile)
     await file.seek(0)
     # If files are in a subfolder of the bucket, prepend the folder path
     bucket_key = resolve_bucket_key(file_name)
 
-    upload_success = await bucket_service.upload_file(bucket_key=bucket_key,
-                                                      file_binary=file.file)
-    if upload_success is False:
-        raise HTTPException(
-            status_code=500,
-            detail="The upload did not succeed"
-        )
-    entry = dict(**entry)
-    entry["bucket_key"] = bucket_key
-    return await crud.update_entry(media, MediaCreation(**entry), media_id)
+    # Upload if bucket_key is different (otherwise the content is the exact same)
+    if isinstance(entry['bucket_key'], str) and entry['bucket_key'] == bucket_key:
+        return await crud.get_entry(media, media_id)
+    else:
+        # Failed upload
+        if not await bucket_service.upload_file(bucket_key=bucket_key, file_binary=file.file):
+            raise HTTPException(
+                status_code=500,
+                detail="Failed upload"
+            )
+        # Data integrity check
+        uploaded_file = await bucket_service.get_file(bucket_key=bucket_key)
+        # Failed download
+        if uploaded_file is None:
+            raise HTTPException(
+                status_code=500,
+                detail="The data integrity check failed (unable to download media form bucket)"
+            )
+        # Remove temp local file
+        background_tasks.add_task(bucket_service.flush_tmp_file, uploaded_file)
+        # Check the hash
+        with open(uploaded_file, 'rb') as f:
+            upload_hash = hash_content_file(f.read())
```

Comment:
would suggest
```suggestion
            upload_hash = hash_content_file(open(uploaded_file, 'rb').read())
```

Reply:
Well, here we do want to close the file after hashing, don't we?